core/src/main/scala/cats/data/NonEmptyMapImpl.scala

```diff
@@ -93,6 +93,11 @@ sealed class NonEmptyMapOps[K, A](val value: NonEmptyMap[K, A]) {
    */
   def lookup(k: K): Option[A] = toSortedMap.get(k)
 
+  /**
+   * Applies f to the value stored at k. If lookup misses, does nothing.
+   */
+  def updateWith(k: K, f: A => A): NonEmptyMap[K, A] = lookup(k).fold(value)(v => add((k, f(v))))
```

Comment:
I'm wondering if `updateWith[B >: A](k: K, f: A => B): NonEmptyMap[K, B]` wouldn't be better

Reply:
I like the ability to widen, I also think it's more cats conventional to curry the `f` param.

Reply:
So the underlying collection is covariant in the value type `SortedSet[A, +B]`. I guess it would line up correctly to allow the widening?

Reply:
@kailuowang so just `updateWith(k: K)(f: A => A)`?

Reply:
IDK if this will work because `updateWith` depends on `add` which is in A only...

```
[error]  found   : B
[error]  required: A
[error]   def updateWith[B >: A](k: K)(f: A => B): NonEmptyMap[K, B] = lookup(k).fold(value)(v => add((k, f(v))))
[error]                                                                                                    ^
[error] one error found
```

Reply:
yeah now I realized that  `add` doesn't widen, it doesn't make much sense for updateWith to be able to widen .